capa/ida/helpers.py

```diff
@@ -191,3 +198,33 @@ def read(self, size):
 
     def close(self):
         return
+
+
+def save_results(resdoc, ruleset):
+    n = netnode.Netnode(CAPA_NETNODE)
+    n[NETNODE_RESULTS] = resdoc.json()
+    n[NETNODE_RULESET] = base64.b64encode(pickle.dumps(ruleset)).decode("ascii")
+
+
+def idb_contains_capa_results() -> bool:
+    n = netnode.Netnode(CAPA_NETNODE)
+    return bool(n.get(NETNODE_RESULTS) and n.get(NETNODE_RULESET))
+
+
+def load_results() -> capa.render.result_document.ResultDocument:
+    logger.debug("loading capa results from netnode")
+    n = netnode.Netnode(CAPA_NETNODE)
+    return capa.render.result_document.ResultDocument.parse_obj(json.loads(n[NETNODE_RESULTS]))
+
+
+def load_ruleset() -> CapaExplorerRuleSetCache:
+    logger.debug("loading capa ruleset from netnode")
+    n = netnode.Netnode(CAPA_NETNODE)
+    return pickle.loads(base64.b64decode(n[NETNODE_RULESET]))
```

Comment:
we'll have to be careful with the interplay between the capa ruleset cache and the explorer ruleset cache

Reply:
maybe we should just rely on the disk cached ruleset? its pretty fast and doing this would reduce the chance of weird staleness issues.  just an idea, open to discussion.

Reply:
yes, wanted to wait finalizing this after #1291 is complete

Reply:
> maybe we should just rely on the disk cached ruleset
yes, I think the changes in the other PR lead to that, any way

Reply:
removed here, now only using system rule cache